### PR TITLE
Add rate limiting to cves

### PR DESCRIPTION
**Pull request by samhotep**
## Done

- Added rate limiting to the security and notices APIs

## QA

- Open the demo at https://ubuntu-com-13880.demos.haus/security/cves and open any cve page
- Also open https://ubuntu-com-13880.demos.haus/security/notices and open any notice
- Both should open normally

## Issue / Card

Fixes [WD-11323](https://warthogs.atlassian.net/browse/WD-11323)

[WD-11323]: https://warthogs.atlassian.net/browse/WD-11323?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webapp/context.py [hunk 1]
```diff
@@ -195,3 +199,23 @@ def sort_by_key_and_ordered_list(list_to_sort, obj_key, ordered_list):
...
+    def decorator(fn):
+        def wrapper(*args, **kwargs):
+            limit = parse(request_limit)
+            rate_limit = fixed_window.hit(limit)
+            if not rate_limit:
+                return flask.abort(429, f"The rate limit is: {request_limit}")
+            return fn(*args, **kwargs)
+
+        # Prevent naming conflicts
+        # https://stackoverflow.com/questions/17256602/assertionerror-view-function-mapping-is-overwriting-an-existing-endpoint-functi
```
Comment (andesol) [inline webapp/context.py:217]: Ignore if this is unrelated, but [functools.wraps](https://docs.python.org/3/library/functools.html#functools.wraps) can be used so the name of the function is not lost

**Review by mtruj013 — APPROVED** [review]
Thanks @samhotep , LGTM

**Comment by setharnold** [comment]
Hey, are these rates right? Our website is looking pretty useless all day long. Thanks.

**Comment by carkod** [comment]
There's a decorator version for this library https://flask-limiter.readthedocs.io/en/stable/recipes.html

**Pull request merged**